src/minigames/scrubMinigame.ts

```diff
@@ -0,0 +1,78 @@
+import { Application, Graphics, Point } from "pixi.js";
+import { Minigame } from "./minigame";
+
+interface DirtySpotDelegate {
+  markCleaned(spot: DirtySpot): void;
+}
+
+class DirtySpot {
+  private health = 100;
+  private pointerDown = false;
+
+  constructor(
+    private readonly app: Application,
+    private readonly delegate: DirtySpotDelegate,
+  ) {}
+
+  getObject() {
+    const spot = new Graphics();
+
+    const x = (this.app.screen.width - 100) * Math.random() + 50;
+    const y = (this.app.screen.height - 100) * Math.random() + 50;
+    spot.rect(-50, -50, 100, 100);
+    spot.fill(0xabcdef);
+    spot.position = new Point(x, y);
+
+    spot.eventMode = "static";
+    spot.on("mousemove", (event) => {
+      if (!this.pointerDown) return;
+
+      this.health = Math.max(this.health - event.movement.magnitude() * 0.1, 0);
+      spot.alpha = this.health / 100;
+      if (this.health === 0) {
+        this.delegate.markCleaned(this);
+        spot.removeFromParent();
+      }
+    });
+
+    return spot;

```

Comment:
Huh do we need to return spot?

Reply:
Yes because the spot is added to the container in line 56